crates/symbolicator-service/src/services/caches/downloads.rs

```diff
@@ -0,0 +1,83 @@
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+
+use symbolic::common::ByteView;
+use symbolicator_sources::RemoteFile;
+use tempfile::NamedTempFile;
+
+use crate::caching::{
+    Cache, CacheEntry, CacheItemRequest, CacheKey, CacheVersions, Cacher, SharedCacheRef,
+};
+use crate::services::download::DownloadService;
+use crate::services::fetch_file;
+use crate::types::Scope;
+
+use super::versions::DOWNLOADS_CACHE_VERSIONS;
+
+/// Provides cached access to arbitrary downloaded files, without validating or parsing their contents.
+///
+/// This should be used for any files which are not being validated / converted upon download.
+/// An example might be downloads for individual source files.
+///
+// XXX: The only "validation" that source files might need is checking that they are valid UTF-8.
+// Not sure if its worth validating this at download time, to persist a `Malformed` marker?
```

Comment:
This might be a valid concern. Not sure if this cache might be a bit too generic?

Reply:
I think it's probably reasonable to do UTF-8 validation here; I don't think we anticipate this being used for arbitrary binary files?

Reply:
I'm not sure if it's worth the time to do the validation, but I think either choice are fine.